README.adoc

```diff
@@ -0,0 +1,138 @@
+
+
+== Features
+
+TODO
+
+
+== Installation from Source on CentOS 7
+
+IMPORTANT: Node agent needs to be installed on every node of the storage
+cluster that is to be managed by tendrl.
+
+NOTE: All the commands are run as a regular user that has `sudo` privileges.
+The commands are all assumed to be run from a single directory, which by
+default could be the user's home directory.
+
+=== Deployment Requirements
+
+. Ensure that etcd is running on a node in the network and is reachable from
+the node you're about to install the node agent on. Note it's address and port.
+
+=== System Setup
+
+. Install the build toolchain and other development packages.
+
+ $ sudo yum groupinstall 'Development Tools'
+ $ sudo yum install libffi-devel openssl-devel
+
+. Install https://virtualenvwrapper.readthedocs.io/[virtualenvwrapper].
+
+ $ sudo yum install python-virtualenv python-virtualenvwrapper
+
+. Create system directories.
+
+ $ sudo mkdir -p /etc/tendrl /var/log/tendrl/node_agent
+
+=== Environment Setup
+
+. Configure `virtualenvwrapper`.
++
+Setup the shell startup files based on the `virtualenvwrapper` documentation
+at:
+https://virtualenvwrapper.readthedocs.io/en/latest/install.html#shell-startup-file
++
+Be sure to adjust the value of `source` to the output of:
+
+ $ which virtualenvwrapper.sh # Should be /usr/bin/virtualenvwrapper.sh
+
+. Create and load the virtual environment for the node agent.
+
+ $ mkvirtualenv tendrl-node-agent
+ $ workon tendrl-node-agent
+
+=== Install bridge_common and node_agent
+
+. Install the https://github.com/Tendrl/common[bridge common library].
+
+ $ git clone https://github.com/Tendrl/common.git bridge_common
+ $ pushd bridge_common
+ [bridge_common] $ workon tendrl-node-agent
+ [bridge_common] $ pip install .
+ [bridge_common] $ popd
+
+. Install the node agent.
+
+ $ git clone https://github.com/Tendrl/node_agent.git
+ $ pushd node_agent
+ [node_agent] $ workon tendrl-node-agent
+ [node_agent] $ pip install .
+ [node_agent] $ popd
+
+=== Configuration
+
+. Create the tendrl configuration file `/etc/tendrl/tendrl.conf`.
+
+ $ cp bridge_common/etc/tendrl/tendrl.conf.sample /etc/tendrl/tendrl.conf
+
+.. Configure the following `etcd_port` and `etcd_connection` directives in
+`/etc/tendrl/tendrl.conf` to point to the etcd instance discussed in the first
+step.
+
+. Install the bridge_common logging configuration file
+`/etc/tendrl/common_logging.yaml`.
+
+ $ cp bridge_common/etc/samples/logging.yaml.timedrotation.sample \
+   /etc/tendrl/common_logging.yaml
+
+NOTE: There are other sample configuration files in the
+`bridge_common/etc/samples` directory which could be used to setup logging for
+different system configuration such as via syslog and journald.
+
+. Install the node agent logging configuration file
+`/etc/tendrl/node_agent_logging.yaml`.
+
+ $ cp node_agent/etc/logging.yaml.timedrotation.sample \
+   /etc/tendrl/node_agent_logging.yaml
+
```

Comment:
You need edit the copied file appropriately so that the logs go into the correct folder. logging.yaml.timedrotation.sample always points to /var/log/tendrl/node_agent/node_agent*.